errors.go

```diff
@@ -218,7 +218,8 @@ func (p *Error) Retryable() bool {
 }
 
 // Unexpected states whether an error is not expected to occur. In many cases this will be due to a bug, e.g. due to a
-// defensive check failing
+// defensive check failing.
+// Note that if the IsUnexpected flag has not been set at all, this will still return false.
```

Comment:
Felt it was worth adding a note here. Not changing this means this change should have no practical impact for someone relying on this function (we defaulted to false -> false, now we default to nil -> false).

Longer term I question if we really want this function, or perhaps we should have an equivalent `Expected()` function (that returns true only if the flag is set to false). But that risks making this more of a breaking change.